In [7]:
%matplotlib inline
import numpy as np
import pandas as pd
import scipy.constants as const
import matplotlib.pyplot as plt
from scipy.interpolate import interp1d

# Problem 1

Define given values

In [5]:
flux = 1e9 # cm^-2 s^-1
energy = 100 # eV
thermal_energy = 0.025 # eV

## Part A

Calculate absorption cross section using thermal $\sigma_a=98.7b$

Cross section at low energies goes as $\sigma \propto \frac{1}{v} \propto \sqrt{\frac{1}{T}}$. Therefore: $$ \sigma^{100eV}_a \approx \sigma^{0.025eV}_a \times \sqrt{\frac{0.025eV}{100eV}} $$

In [6]:
98.7 * np.sqrt(thermal_energy / energy)

1.5605840252930951

$$ \sigma_a^{100eV} \approx 1.56b $$

ENDF/B-VII.1 gives values of $\sigma_a(99.579eV) = 3.90669b$ and $\sigma_a(100.246eV) = 3.99648b$ for the $(n,\gamma)$ reaction. Interpolating between these:

In [12]:
crosssection = interp1d(x=[99.579, 100.246], y=[3.90669, 3.99648])
print(crosssection(100))
print(crosssection(100) / (98.7 * np.sqrt(thermal_energy / energy)))

3.96336404798
2.53966719109


This is a factor of 2.54 higher than the calculated value. If we examine the plot of the $(n, \gamma)$ cross section we see:
![cross section graph](